litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go

```diff
@@ -88,3 +88,50 @@ func NewEvent(clusterEvent model.ClusterEventInput, r store.StateData) (string,
 
 	return "", errors.New("ERROR WITH CLUSTER EVENT")
 }
+
+// UpsterWorkFlowRun Updates or Inserts a new Workflow Run into the DB
+func UpsterWorkFlowRun(input model.WorkflowRunInput, r store.StateData) (string, error) {
+	cluster, err := cluster.VerifyCluster(*input.ClusterID)
+	if err != nil {
+		log.Print("ERROR", err)
+		return "", err
+	}
+	newWorkflowRun := model.WorkflowRun{
```

Comment:
https://github.com/litmuschaos/litmus/pull/1841/files#r467488791

Reply:
Yes here I am using the WorkflowRun struct and using the same instance for bth gql and mongo operations at line 109 and 110. If we separate the 2 structs I will have to repeat this block of code again

Reply:
You don't have to repeat the struct. You can simply convert one struct to another.
Ref- https://github.com/litmuschaos/litmus/blob/litmus-portal/litmus-portal/backend/graphql-server/graph/schema.resolvers.go#L95
here I converted verifiedCluster struct to gql struct

Reply:
Okay, I'll try to convert it then. But having the same struct twice might not be an efficient way of doing this.

Reply:
Yes, we can do this if the gql struct is not autogenerated. We don't have to change autogen struct.